src/lib.rs

```diff
@@ -0,0 +1,93 @@
+use lazy_static::lazy_static;
+use std::collections::HashSet;
+
+use guest::prelude::*;
+use kubewarden_policy_sdk::wapc_guest as guest;
+
+use k8s_openapi::api::core::v1 as apicore;
+
+extern crate kubewarden_policy_sdk as kubewarden;
+use kubewarden::{logging, protocol_version_guest, request::ValidationRequest, validate_settings};
+
+mod settings;
+use settings::Settings;
+
+use slog::{error, o, warn, Logger};
+
+lazy_static! {
+    static ref LOG_DRAIN: Logger = Logger::root(
+        logging::KubewardenDrain::new(),
+        o!("policy" => "priority-class-policy")
+    );
+}
+
+#[no_mangle]
+pub extern "C" fn wapc_init() {
+    register_function("validate", validate);
+    register_function("validate_settings", validate_settings::<Settings>);
+    register_function("protocol_version", protocol_version_guest);
+}
+
+fn validate(payload: &[u8]) -> CallResult {
+    let validation_request: ValidationRequest<Settings> = ValidationRequest::new(payload)?;
+    match validation_request.extract_pod_spec_from_object() {
+        Ok(Some(pod_spec)) => {
+            match validate_pod_priority_class(
+                pod_spec,
+                validation_request.settings.allowed_priority_classes,
+            ) {
+                Ok(_) => kubewarden::accept_request(),
+                Err(err) => kubewarden::reject_request(Some(err.to_owned()), None, None, None),
+            }
+        }
+        Ok(None) => {
+            warn!(LOG_DRAIN, "no PodSpec found");
+            kubewarden::accept_request()
+        }
+        Err(_) => {
+            error!(LOG_DRAIN, "cannot extract PodSpec");
+            kubewarden::accept_request()
+        }
+    }
+}
+
+fn validate_pod_priority_class(
+    pod: apicore::PodSpec,
+    allowed_priority_classes: HashSet<String>,
+) -> Result<(), String> {
+    if let Some(priority_class_name) = pod.priority_class_name {
+        if !allowed_priority_classes.contains(&priority_class_name) {
+            return Err(format!(
+                "Priority class \"{}\" is not allowed",
+                priority_class_name
+            ));
+        }
+    }
+    Ok(())
```

Comment:
A matter of personal taste, I don't like too many nested `ifs`:

I would do something like that:
```rust
if pod.priority_class_name.is_none() {
  return Ok(())
}

let priority_class_name = pod.priority_class_name.unwrap();
// ...

```